app/src/sharedTest/java/org/oppia/app/player/state/StateSelectionInteractionTest.kt

```diff
@@ -0,0 +1,120 @@
+package org.oppia.app.player.state
+
+import android.app.Activity
+import android.app.Application
+import android.content.Context
+import android.content.Intent
+import androidx.test.core.app.ActivityScenario
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.ActivityTestRule
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import junit.framework.Assert.assertTrue
+import kotlinx.coroutines.CoroutineDispatcher
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.app.home.HomeActivity
+import org.oppia.app.recyclerview.RecyclerViewMatcher.Companion.atPosition
+import org.oppia.util.threading.BackgroundDispatcher
+import org.oppia.util.threading.BlockingDispatcher
+import javax.inject.Singleton
+
+/** Tests for SelectionInteraction [MULTIPLE_CHOICE_INPUT], [ITEM_SELECT_INPUT]. */
+@RunWith(AndroidJUnit4::class)
+class StateSelectionInteractionTest {
+
+  private val maxSelectionAllowedCount = 2
+  private var counter = 0
+  private lateinit var launchedActivity: Activity
+
+  @get:Rule
+  var activityTestRule: ActivityTestRule<HomeActivity> = ActivityTestRule(
+    HomeActivity::class.java, /* initialTouchMode= */ true, /* launchActivity= */ false
+  )
+
+  @Before
+  fun setUp() {
+    Intents.init()
+    val intent = Intent(Intent.ACTION_PICK)
+    launchedActivity = activityTestRule.launchActivity(intent)
+  }
+
+  @Test
+  fun testMultipleChoiceInput_showsRadioButtons_forDemoExploration_withCustomOppiaTags_userSelectsDesiredOption() {
```

Comment:
Can these test-cases check whether the radio-button has been displayed or checkbox has been displayed? I think we should check that too.

Reply:
It is based on the exploration interaction Id. Do I need to load exploration here?

These testcases are tested with both exploration Id from demo explorations, so at a time it will load only one.

Reply:
I think you can restrict few test-cases which are based on a specific exploration_id. Those test-cases are important.

Reply:
Ben had suggested me in one of the PR not to include test exploration_id and instead use demoExploration.

Reply:
Okay, then maybe we need to change this approach for testing, but I do think that we should have such test-cases.

@BenHenning Can you help us decide on how to check these test cases:

1. String on the item is displayed correctly.
2. In MultipleChoiceInput, if learner selects one option and then select second option than it should deselect first option.
3. In ItemSelectionInput, if the max-answers-allowed is 1, then radio buttons are visible otherwise checkboxes are visible.
4. In MultipleChoiceInput, after selection one option, if there are configuration changes than the option should remain selected. Similar test case for ItemSelectionInput too.
5. In ItemSelectionInput, if learner selects one option and then selects other option, then both should remain checked.
6. Test to confirm that proper radio-button or checkboxes are visible.
Also, I have faced on issue with checkboxes in recyclerview, if the list it too long, lets say 20 items, then if I select item-2 and scroll to bottom than some-other items will automatically get checked because RecyclerView recyclers the items. So if possible do create such scenario practically with 20 options and then select few options and then scroll and check the behaviour.

Reply:
It would probably be easiest to create custom explorations via a local dev instance of Oppia that sets a single state with an interaction configured as needed for all these different situations, export them to JSON, import them into the project, and hook those up to our ExplorationRetriever with new IDs. Then you can load the custom exploration corresponding to your test scenario to verify the specific situation.

I think that will help cover cases (1)-(5). For case (6), you could create an exploration with lots of checkboxes. This is a bit tricky though because scrolling is a bit tenuous with Robolectric & Espresso, but I think it's worth trying to test this case especially if we anticipate there being issues.